### PR TITLE
support preact X

### LINKED ISSUE #1301: Support to Preact X
Hey.

I observed that there's no support to Preact X yet.

Specifically, the typescript compiler emits an error about code using the `attributes` attribute in the VNode structure, like [this example](https://github.com/prateekbh/preact-material-components/blob/master/ts/Base/MaterialComponent.ts#L170-L178).

I imagine this is related to the fact that in Preact X the VNode shape has changed, and now `attributes` is just `props` in the VNode structure. According to Typescript compiler, there's a few other cases where the same error occurs.

So here's the question: What's the plan to support Preact X? Would this package have a version compatible with both versions (assuming this is the unique problem, of course), or we would release a new version supporting only Preact X?

PS: I know that Preact X is still in alpha but I think it's better to prepare for it than await to a stable release...What you think?

### PR DESCRIPTION
**Pull request by prateekbh**
Fixes #1301

### CODE REVIEW

**Comment by artipost** [comment]
Failed pictures (25): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/c37aa9faec187aaf18e0be76ea06ab52/raw/9635cfcc441b231d89b9de4c19b40606ec086173/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577554944](https://travis-ci.org/prateekbh/preact-material-components/builds/577554944), for commit [9d32a2f](https://github.com/prateekbh/preact-material-components/pull/1334/commits/9d32a2f3ffc7167bf57325c8b4dd84b12bde2540).

**Comment by cromefire** [comment]
Was it intended to be against master?

**Comment by artipost** [comment]
Failed pictures (25): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/622a7d7fe21f93ecd883e13dfae9f525/raw/d54fe80a190aa3ac641544afbca15e5cd7d856b2/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577558618](https://travis-ci.org/prateekbh/preact-material-components/builds/577558618), for commit [c1f70ef](https://github.com/prateekbh/preact-material-components/pull/1334/commits/c1f70efaef872436a03e437aaba1779d6682a303).

File: ts/Base/MaterialComponent.ts [hunk 1]
```diff
@@ -76,9 +76,11 @@ export abstract class MaterialComponent<
     }
 
     const element = this.materialDom(componentProps);
-    element.attributes = element.attributes || {};
+    // @ts-ignore
```
Comment (cromefire) [inline ts/Base/MaterialComponent.ts]: Are the types bad? `@ts-ignore` should be avoided if necessary.

File: ts/Base/MaterialComponent.ts [hunk 2]
```diff
@@ -76,9 +76,11 @@ export abstract class MaterialComponent<
...
+    element.props = element.props || {};
```
Comment (cromefire) [inline ts/Base/MaterialComponent.ts]: This breaks compatibility with preact 8.X.

You could probably extract this into a separate function (like `getProps()`)

**Comment by artipost** [comment]
Failed pictures (25): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/07e5b116b03331f6618e7c84099dc524/raw/7b02dbb095e98f37dcb57fdec9b709609833d9aa/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577561065](https://travis-ci.org/prateekbh/preact-material-components/builds/577561065), for commit [961297f](https://github.com/prateekbh/preact-material-components/pull/1334/commits/961297f444585417f0a1e48f91a98761f8ed09be).

**Comment by artipost** [comment]
Failed pictures (25): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/5c2c73dc16a3953df0746421b53ecb8a/raw/1c5c5a408f32133199dc1ec76a2a40cb02e44a81/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577562877](https://travis-ci.org/prateekbh/preact-material-components/builds/577562877), for commit [802354b](https://github.com/prateekbh/preact-material-components/pull/1334/commits/802354b467c6fc1f4579b0fa6eee4169f3a97771).

File: docs/package.json [hunk 3]
```diff
@@ -46,7 +46,10 @@
     "mocha": "^5.0.1",
     "node-sass": "^4.9.3",
     "node-static": "^0.7.11",
+    "preact": "^10.0.0-rc.1",
     "preact-cli": "^3.0.0-next.14",
+    "preact-render-to-string": "^4.1.0",
```
Comment (cromefire) [inline docs/package.json]: This has only to be fixed with the new `preact-cli` version (in should be in `2.0`)

File: ts/Base/MaterialComponent.ts (hunk 2 above)
Comment (cromefire) [reply ts/Base/MaterialComponent.ts]: preact 8.X has no `props` right?

File: ts/Base/MaterialComponent.ts (hunk 1 above)
Comment (cromefire) [reply ts/Base/MaterialComponent.ts]: [The types have `props` defined](https://github.com/preactjs/preact/blob/4a581a1c4750ca5d71e416deaa3bdf3061b9dbe5/src/index.d.ts#L15), are you sure, there aren't bad types lingering around?

**Comment by cromefire** [comment]
Support for both is quite easy: If there is `props` use that else use attributes

**Comment by artipost** [comment]
Failed pictures (25): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/e1575e98ce8dc72da9727db8a12e5dca/raw/87893ca873636dfb520cb3f414616d6ed166a6a8/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577576309](https://travis-ci.org/prateekbh/preact-material-components/builds/577576309), for commit [abf5e05](https://github.com/prateekbh/preact-material-components/pull/1334/commits/abf5e050bebc524dcdf056a034ca4c60abdaf122).

**Comment by artipost** [comment]
Failed pictures (25): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/6f8e7a2db4a96ab2ff3fcdbcdaad348a/raw/a7c81d567066bda1859b03b21afe54367785bc6c/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577580232](https://travis-ci.org/prateekbh/preact-material-components/builds/577580232), for commit [979ce1b](https://github.com/prateekbh/preact-material-components/pull/1334/commits/979ce1bbece8ace5cd73da35793e9fe5d71327ed).

**Comment by artipost** [comment]
Failed pictures (24): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/ac95c534526537814594d4fcc8178e2c/raw/7357a492f0371a6730573b22a9315bd2e1c77fda/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577586818](https://travis-ci.org/prateekbh/preact-material-components/builds/577586818), for commit [8c80489](https://github.com/prateekbh/preact-material-components/pull/1334/commits/8c804899014319d3bcac7a413760097bd136ddfb).

**Comment by artipost** [comment]
Failed pictures (25): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/b6aa8de2931f3da63aa5431e993a7bd5/raw/baac47497c5c1ab7b854d6463b7b77d87b61e1da/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577591433](https://travis-ci.org/prateekbh/preact-material-components/builds/577591433), for commit [dcd4df2](https://github.com/prateekbh/preact-material-components/pull/1334/commits/dcd4df23a2ba64ccdbf50d2cc4c6fbe2213b7b8b).

**Comment by artipost** [comment]
Failed pictures (24): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/d142ae8efd65bda163e4a26be1908143/raw/6799ce084d0135a78f6a57653eeff374157e033d/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577594052](https://travis-ci.org/prateekbh/preact-material-components/builds/577594052), for commit [b71355e](https://github.com/prateekbh/preact-material-components/pull/1334/commits/b71355e2f91d463f86a1ab5d6b3b0db3437125aa).

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/bcdcbd00c79a6b0dc00beee661abc1ad/raw/5f80053f71ff6e0fade01e1033ede941a05fd574/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [577596174](https://travis-ci.org/prateekbh/preact-material-components/builds/577596174), for commit [56f6a0c](https://github.com/prateekbh/preact-material-components/pull/1334/commits/56f6a0c5329d99b51f8075a28aef32246e07f024).

**Pull request merged**